### ISSUE TITLE
External reference links (MSDN) do not resolve correctly

### ISSUE THREAD

**Issue by qJake** [Geaz/SharpDox.Plugins.Html#21]
If I write documentation like this:

```csharp
using System.IO;

/// <summary>
/// Writes a file using the <see cref="File" /> class.
/// </summary>
public void WriteFile() { ... }
```

The link that gets generated for that is a local link - its href is `http://mydocsite.com/#/type/System.IO.File/index`. This does not resolve correctly.

Instead, the system should scan through all known namespaces/references, and if it cannot find a match, it should instead link (with `target="_blank"` to open in a new window) to MSDN.

MSDN links are easy to generate, I believe Microsoft will resolve anything after `/lang-code/library/...`, for example, to link to the File class in English, it is simply: [https://msdn.microsoft.com/en-us/library/System.IO.File](https://msdn.microsoft.com/en-us/library/System.IO.File)

Or, for `WebClient.DownloadString()` in German: [https://msdn.microsoft.com/de-de/library/System.Net.WebClient.DownloadString](https://msdn.microsoft.com/de-de/library/System.Net.WebClient.DownloadString)

If there is a concern about linking to *all* external libraries (for example, you would not want to generate a link to MSDN for `Newtonsoft.Json.Linq.JObject`), you could filter on just `System.*` and `Microsoft.*`.

**Comment by geaz** [comment]
Yes, sharpDox does not support external references at the moment. I don't know, if sharpDox will get this feature in its current state, because I am writing a new version of sharpDox for .NET Core at the moment.

**Comment by qJake** [comment]
If this feature is not already in coreDox too, you should add it! 😃
